sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -862,7 +862,17 @@ class Analyzer(override val catalogManager: CatalogManager)
   }
 
   private def isResolvingView: Boolean = AnalysisContext.get.catalogAndNamespace.nonEmpty
-  private def referredTempViewNames: Seq[Seq[String]] = AnalysisContext.get.referredTempViewNames
+  private def isReferredTempViewName(nameParts: Seq[String]): Boolean = {
+    AnalysisContext.get.referredTempViewNames.exists { n =>
+      (n.length == nameParts.length) && n.zip(nameParts).forall {
+        case (a, b) => resolver(a, b)
+      }
+    }
+  }
+
+  private def getTempViewRawPlan(plan: LogicalPlan): LogicalPlan = {
+    EliminateDataFrameTempViews(EliminateSubqueryAliases(plan))
```

Comment:
`EliminateDataFrameTempViews` doesn't need to be a rule anymore, right? we can probably just do 
```
EliminateSubqueryAliases(plan) match {
  case v: View if v.isDataFrameTempView => v.child
  case other => other
}
```